### PR TITLE
Move diagnosis and reason for visit to new profile

### PR DESCRIPTION
**Pull request by mikajylha**
Creates a "reason for care" profile. Moves diagnosis and käyntisyy stuff there.

Overview of changes:
- Move all condition intro markdown content to reason-for-care intro
- Almost completely rewrites Scope and Usage from both intros
- Remove FiBaseConditionCategory as it's not needed anymore
-  Change rules for how encounter-diagnosis category is interpreted

Alters naming and wording so that "reason for visit" refers to THL käyntisyy and the more general "reason for care" to both diagnosis and käyntisyy. This should align concepts very closely to the THL's Tiedonhallintapalvelu -spec.

### CODE REVIEW

File: input/fsh/reasonForCare.fsh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+Profile: FiBaseReasonForCare
+Parent: FiBaseCondition
+Id: fi-base-reason-for-care
+Title: "FI Base Reason for Care (encounter-diagnosis and reason for visit)"
+Description: "This is the Finnish base profile for the encounter-diagnosis and (nurse asserted) reason for visit Conditions."
```
Comment (mrinnetmaki) [inline input/fsh/reasonForCare.fsh:5]: I think this is still a bit hard to read... But I don't have a better wording to offer right now.

File: input/pagecontent/StructureDefinition-fi-base-condition-intro.md [hunk 2]
```diff
@@ -1,225 +1,7 @@
...
-    * map to [verificationStatus codes](https://hl7.org/fhir/R4/valueset-condition-ver-status.html)?
-* THL Tietosisältö 9 Diagnoosin tai käyntisyyn episoditunnus
-
-#### Links
+FiBaseReasonForCare SHOULD be used for  Diagnosis -based conditions and Finnish *kayntisyy*
+conditions instead of this.
 
-* [Suomalainen tautien kirjaamisen ohjekirja](https://thl.fi/documents/10531/124365/Opas%202012%2017.pdf)
-national guide for the use if ICD-10.
+This profile may be used for describing reason given by patient for requesting/acquiring healthcare
```
Comment (mrinnetmaki) [inline input/pagecontent/StructureDefinition-fi-base-condition-intro.md:6]: MAY?

File: input/fsh/condition.fsh
[the remark this answers is not in the document]
Comment (mikajylha) [reply input/fsh/condition.fsh:1]: Fixing...

[the remark this answers is not in the document]
Comment (mikajylha) [reply input/fsh/condition.fsh:1]: Fixed

**Pull request merged**